### PR TITLE
[cleanup] [broker] fix return value of TransactionTimeoutTrackerImpl#addTransaction

### PR DESCRIPTION
**Pull request by thetumbled**
### Motivation
According to the method signature, we should return true when the transaction is added to the tracker.
```
    /**
     * Add a txnID to the tracker.
     *
     * @param sequenceId
     *            the sequenceId
     * @param timeout
     *            the absolute timestamp for transaction timeout
     *
     * @return true if the transaction was added to the tracker or false if had timed out
     */
    CompletableFuture<Boolean> addTransaction(long sequenceId, long timeout);
```
But actually, we return false for any cases.

Update: Moreover, we do not use the return value anyway, it is useless. We have better remove it.

### Modifications

~~Return true when the transaction is added to the tracker.~~
Remove the return value as it is useless.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/thetumbled/pulsar/pull/41

### CODE REVIEW

**Comment by liangyepianzhou** [comment]
Hi, @thetumbled Thanks for this catch.
IMO, It is better to delete the return value.

**Review by liangyepianzhou — APPROVED** [review]

**Comment by Technoboy-** [comment]
> Hi, @thetumbled Thanks for this catch. IMO, It is better to delete the return value.

This method has existed since 2.8. We can't change this without discussion.

**Comment by liangyepianzhou** [comment]
>This method has existed since 2.8. We can't change this without discussion.

I agree. Even though this is a Beta interface, it has been around for too long. We can start a discussion.

**Comment by thetumbled** [comment]
The discussion is at https://lists.apache.org/thread/hf3m2f0lxbksso761wbowcynotn45kbr

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/timeout/TransactionTimeoutTrackerImpl.java [hunk 1]
```diff
@@ -57,10 +56,9 @@ public class TransactionTimeoutTrackerImpl implements TransactionTimeoutTracker,
     }
 
     @Override
-    public CompletableFuture<Boolean> addTransaction(long sequenceId, long timeout) {
+    public void addTransaction(long sequenceId, long timeout) {
         if (timeout < tickTimeMillis) {
             this.transactionMetadataStoreService.endTransactionForTimeout(new TxnID(tcId, sequenceId));
-            return CompletableFuture.completedFuture(false);
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/timeout/TransactionTimeoutTrackerImpl.java:63]:
```java
- return CompletableFuture.completedFuture(false);
+ return;
```

**Review by poorbarcode — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/timeout/TransactionTimeoutTrackerImpl.java (hunk 1 above)
Comment (thetumbled) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/timeout/TransactionTimeoutTrackerImpl.java:63]: thanks for catch, fixed.

**Comment by poorbarcode** [comment]
> According to the method signature, we should return true when the transaction is added to the tracker.
> ```
>     /**
>      * Add a txnID to the tracker.
>      *
>      * @param sequenced
>      *            the sequenced
>      * @param timeout
>      *            the absolute timestamp for transaction timeout
>      *
>      * @return true if the transaction was added to the tracker or false if had timed out
>      */
>     CompletableFuture<Boolean> addTransaction(long sequenceId, long timeout);
> ```
> But actually, we return false for any cases.

You changed the definition to `void addTransation()`, it is not matches with the Motivation, please correct it.

**Comment by thetumbled** [comment]
> > According to the method signature, we should return true when the transaction is added to the tracker.
> > ```
> >     /**
> >      * Add a txnID to the tracker.
> >      *
> >      * @param sequenced
> >      *            the sequenced
> >      * @param timeout
> >      *            the absolute timestamp for transaction timeout
> >      *
> >      * @return true if the transaction was added to the tracker or false if had timed out
> >      */
> >     CompletableFuture<Boolean> addTransaction(long sequenceId, long timeout);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But actually, we return false for any cases.
> 
> You changed the definition to `void addTransation()`, it is not matches with the Motivation, please correct it.

Updated, thanks.

**Review by poorbarcode — APPROVED** [review]

**Pull request merged**